src/main/kotlin/com/openlattice/graph/Graph.kt

```diff
@@ -356,8 +382,16 @@ class Graph(private val hds: HikariDataSource, private val edm: EdmManager) : Gr
                 filteredRankings.mapIndexed { index, authorizedFilteredRanking ->
                     val tableSql = buildAssociationTable(index, entitySetIds, authorizedFilteredRanking, linked)
                     //We are guaranteed at least one association for a valid top utilizers request
+
                     if (index == 0) {
-                        "SELECT *,$scoreColumn FROM ($tableSql) as assoc_table$index "
+                        val selectedColumns =
+                                if( linked ) {
+                                    " *, UUID('${linkingEntitySetId.get()}') as $SELF_ENTITY_SET_ID, ${LINKING_ID.name} as $SELF_ENTITY_KEY_ID, $scoreColumn "
```

Comment:
Shouldn't you alias linking id to SELF_ENTITY_KEY_ID?

Reply:
It is: ${LINKING_ID.name} as $SELF_ENTITY_KEY_ID

Reply:
Ah, for some reason it got cut off by github. LGTM.